Detectors/TRD/workflow/src/TRDGlobalTrackingSpec.cxx

```diff
@@ -370,6 +371,43 @@ void TRDGlobalTracking::run(ProcessingContext& pc)
   std::iota(trackIdxArray.begin(), trackIdxArray.end(), 0);
   std::sort(trackIdxArray.begin(), trackIdxArray.end(), [tracksOutRaw](int lhs, int rhs) { return tracksOutRaw[lhs].getCollisionId() < tracksOutRaw[rhs].getCollisionId(); });
 
+  std::vector<std::pair<uint8_t, uint8_t>> pileUpDist;
+  bool ft0Seen = false;
+  if (mTrkMask[GTrackID::FT0]) { // pile-up tagging was requested
+    long maxDiffFwd = mTracker->Param().rec.trd.pileupFwdNBC;
+    long maxDiffBwd = mTracker->Param().rec.trd.pileupBwdNBC;
+    auto ft0recPoints = inputTracks.getFT0RecPoints();
+    auto trdTriggers = tmpInputContainer->mTriggerRecords;
+    ft0Seen = ft0recPoints.size() > 0;
+    pileUpDist.resize(trdTriggers.size(), {0, 0});
+    size_t curFT0 = 0;
+    for (size_t itrd = 0; itrd < trdTriggers.size(); itrd++) {
+      const auto& trig = trdTriggers[itrd];
+      uint8_t fwd = 0, bwd = 0;
+      for (size_t ft0id = curFT0; ft0id < ft0recPoints.size(); ft0id++) {
+        const auto& f0rec = ft0recPoints[ft0id];
+        if (f0rec.getTrigger().getVertex()) {
+          auto bcdiff = trig.getBCData().toLong() - f0rec.getInteractionRecord().toLong();
+          if (bcdiff > maxDiffBwd) {
+            curFT0 = ft0id + 1;
+            continue;
+          }
+          if (bcdiff > 0) { // pre-trigger pileup, maxDiffBwd is guaranteed to be < max uint8_t
+            if (bwd == 0) {
```

Comment:
Should this condition not be removed? Such that in case there are multiple pre-trigger pileup collision the closest one is chosen and not only the first one?

Reply:
No, we want to find the pre- and post- trigger pile-ups with the largest acceptable distance to the trigger.
The FT0 recpoints are orderer so when accounting pre-trigger pile-up and when the `bwd` is not modified yet (==0) the 1st encountered FT0 recpoint within the `maxDiffBwd` from the trigger gives the pre-trigger pile-up we need, we don't need to update `bwd` further.